cypress/e2e/registration.cy.js

```diff
@@ -1,11 +1,62 @@
-/// <reference types='cypress' />
-
-describe('Student Registration page', () => {
-  before(() => {
-
-  });
-
-  it('', () => {
-
-  });
-});
+/// <reference types='cypress' />
+
+describe('Student Registration page', () => {
+  const firstName = 'John';
+  const lastName = 'Doe';
+  const email = 'john.doe@gmail.com';
+  const mobileNumber = '0966313444';
+  const currentAddress = 'Wilhelm-Kalle-Straße 18';
+
+  before(() => {
+    cy.visit('https://demoqa.com/automation-practice-form');
+  });
+
+  it(`should fill all fields except "picture"`, () => {
+    // Fill in the form fields
+    cy.get('#firstName').type(firstName);
+    cy.get('#lastName').type(lastName);
+    cy.get('#userEmail').type(email);
+    cy.get('label[for="gender-radio-1"]').click();
+    cy.get('#userNumber').type(mobileNumber);
+    cy.get('.subjects-auto-complete__value-container')
+      .type('Maths{enter}');
+    cy.get('#currentAddress').type(currentAddress);
+
+    // Select date of birth
+    cy.get('#dateOfBirthInput').click();
+    cy.get('.react-datepicker__month-select').select('February');
+    cy.get('.react-datepicker__year-select').select('1986');
+    cy.get('.react-datepicker__day--020').click();
+
+    // Select hobbies
+    cy.get('label[for="hobbies-checkbox-2"]').click();

```

Comment:
The test selects 'Music' as a hobby, but the assertion checks for 'Reading'. Ensure that the selected hobby matches the expected value in the assertion.